Unreferenced variable warnings

Currently there are a number of unreferenced variable warnings:
```
src/TilePalette.cpp:30:6: warning: unused variable 'yPosition' [-Wunused-variable]
        int yPosition = rect().y() + rect().height() - 23;
            ^
src/TilePalette.cpp:44:12: warning: unused variable 'r' [-Wunused-variable]
        Renderer& r = Utility<Renderer>::get();
                  ^
src/Map/../Stream/StreamWriter.h:13:30: warning: private field 'mStream' is not used [-Wunused-private-field]
        NAS2D::File::RawByteStream              mStream;
                                                ^
src/TextField.cpp:12:11: warning: unused variable 'MIN_WIDTH' [-Wunused-const-variable]
const int MIN_WIDTH             = 16;
          ^
src/TextField.cpp:13:11: warning: unused variable 'MIN_HEIGHT' [-Wunused-const-variable]
const int MIN_HEIGHT    = 16;
          ^
src/TextField.cpp:15:11: warning: unused variable 'MAX_WIDTH' [-Wunused-const-variable]
const int MAX_WIDTH             = 512;
          ^
src/TextField.cpp:16:11: warning: unused variable 'MAX_HEIGHT' [-Wunused-const-variable]
const int MAX_HEIGHT    = 32;
          ^
src/TextField.cpp:18:11: warning: unused variable 'REPEAT_DELAY' [-Wunused-const-variable]
const int REPEAT_DELAY  = 500;
          ^
src/TextField.cpp:19:11: warning: unused variable 'REPEAT_WAIT' [-Wunused-const-variable]
const int REPEAT_WAIT   = 20;
          ^
src/TextField.h:78:10: warning: private field 'mMaxScrollOffset' is not used [-Wunused-private-field]
        int                                     mMaxScrollOffset;       /**< Maximum allowable offset. */
                                                ^
src/EditorState.cpp:131:12: warning: unused variable 'r' [-Wunused-variable]
        Renderer& r = Utility<Renderer>::get();
                  ^
src/EditorState.h:88:9: warning: private field 'mPlacingCollision' is not used [-Wunused-private-field]
        bool                    mPlacingCollision = false;      /**< Flag indicating whether or not to place or clear collision on mouse drags. */
                                ^
src/StartState.cpp:63:17: warning: unused variable 'c' [-Wunused-variable]
        Configuration& c = Utility<Configuration>::get();
                       ^
src/StartState.cpp:86:12: warning: unused variable 'r' [-Wunused-variable]
        Renderer& r = Utility<Renderer>::get();
                  ^
src/StartState.cpp:303:6: warning: unused variable 'mapWidth' [-Wunused-variable]
        int mapWidth = 0, mapHeight = 0;
            ^
src/StartState.cpp:303:20: warning: unused variable 'mapHeight' [-Wunused-variable]
        int mapWidth = 0, mapHeight = 0;
                          ^
src/StartState.cpp:314:17: warning: unused variable 'c' [-Wunused-variable]
        Configuration& c = Utility<Configuration>::get();
                       ^
src/Stream/StreamReader.cpp:3:11: warning: unused variable 'INT_4BYTE_SIZE' [-Wunused-const-variable]
const int       INT_4BYTE_SIZE = 4;
                ^
src/TileGroups.cpp:28:12: warning: unused variable 'r' [-Wunused-variable]
        Renderer& r = Utility<Renderer>::get();
                  ^
```

--- a/src/StartState.cpp
+++ b/src/StartState.cpp
@@ -305,8 +305,6 @@ void StartState::button_CreateNew_click()
 		return;
 	}
 
-	Configuration& c = Utility<Configuration>::get();
-
 	std::string mapname = txtMapPath.text();
 	// check for file extension
 	if (mapname.length() > 4)

--- a/src/TextField.cpp
+++ b/src/TextField.cpp
@@ -9,15 +9,6 @@
 
 #include <locale>
 
-const int MIN_WIDTH		= 16;
-const int MIN_HEIGHT	= 16;
-
-const int MAX_WIDTH		= 512;
-const int MAX_HEIGHT	= 32;
-
-const int REPEAT_DELAY	= 500;
-const int REPEAT_WAIT	= 20;
-
 
 const int FIELD_PADDING = 4;
 

--- a/src/TextField.cpp
+++ b/src/TextField.cpp
@@ -19,7 +19,6 @@ std::locale LOC;
 TextField::TextField():	mCursorPosition(0),
 						mCursorX(0),
 						mScrollOffset(0),
-						mMaxScrollOffset(0),
 						mMaxCharacters(0),
 						mBorderVisibility(FOCUS_ONLY),
 						mEditable(true),

--- a/src/TextField.h
+++ b/src/TextField.h
@@ -75,7 +75,6 @@ private:
 	int 				mCursorPosition;	/**< Position of the Insertion Cursor. */
 	int 				mCursorX;			/**< Pixel position of the Cursor. */
 	int 				mScrollOffset;		/**< Scroller offset. */
-	int					mMaxScrollOffset;	/**< Maximum allowable offset. */
 
 	size_t				mMaxCharacters;		/**< Max number of characters allowed in the text field. */
 

--- a/src/TilePalette.cpp
+++ b/src/TilePalette.cpp
@@ -27,8 +27,6 @@ TilePalette::~TilePalette()
 
 void TilePalette::_init()
 {
-	int yPosition = rect().y() + rect().height() - 23;
-
 	Renderer& r = Utility<Renderer>::get();
 	_rect() = {r.width() - PALETTE_DIMENSIONS.x() - 2, r.height() - PALETTE_DIMENSIONS.y() - 2, static_cast<float>(PALETTE_DIMENSIONS.x()), static_cast<float>(PALETTE_DIMENSIONS.y())};
 
@@ -41,8 +39,6 @@ void TilePalette::_init()
  */
 void TilePalette::draw()
 {
-	Renderer& r = Utility<Renderer>::get();
-
 }
 
 

